taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,351 @@
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            newNodeObject[key] = [];
+            return newNodeObject
+        })
+        setOpen(false);
+    }
+
+    // Used for deleting a translation language
+    function handleDeleteTranslation(key) {
+        const newRenderedTranslations = renderedTranslations.filter(obj => !(key === obj.languageCode))
+        setRenderedTranslations(newRenderedTranslations);
+        key = 'tags_' + key; // LC must have a prefix "tags_"
+        
+        // Make changes to the parent NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            delete newNodeObject[key];
+            return newNodeObject
+        })
+        setOpen(false);
+    }
+
+    // Changes the translations to be rendered
+    // Dependent on changes occuring in "originalNodeObject"
+    useEffect(() => {
+
+        // Main langauge tags are considered separately, since they have to be rendered first
+        let mainLangTags = []
+        let otherLangTags = []
+
+        Object.keys(originalNodeObject).forEach((key) => {
+        
+            // Get all tags and its corresponding language code
+            // Tagids need to be recomputed, so shouldn't be rendered
+            // Eg: tags_fr
+        
+            if (key.startsWith('tags') && !key.includes('ids')) {
+
+                // If tags are not for the main language, add them to otherLangTags
+                if (!key.endsWith(originalNodeObject.main_language)) {
 
-                // Slice the language code
-                let languageCode = key.slice(-2);
-                renderedTranslations[languageCode] = nodeObject[key]
+                    // Slice the language code
+                    let languageCode = key.slice(-2);
+                    let tobeInsertedObj = {'languageCode' : languageCode, 'tags' : []} // General format for storing tags for different lc's
+                    originalNodeObject[key].map((tag) => (
+                        tobeInsertedObj['tags'].push({
+                            'index' : uuid.v4(), // Give a unique identifier for each tag
+                            'tag' : tag
+                        })
+                    ))
+                    otherLangTags.push(tobeInsertedObj);
+                }
+                // If tags are for main language, add them to mainLangTags
+                else {
+                    originalNodeObject["tags_"+originalNodeObject['main_language']].map((tag) => (
+                        mainLangTags.push({ 
+                            'index' : uuid.v4(),
+                            'tag' : tag
+                        })
+                    ))
+                }
             }
-    })
-    
+        })
+
+        // Set states
+        setMainLang_renderedTranslations(mainLangTags);
+        setRenderedTranslations(otherLangTags);
+    }, [originalNodeObject]);
```

Comment:
Please check the comment above on this but, I was wondering what's the need for multiple states derived one from the other? and if it may be better to only use the `nodeObject` as it is a state and not a derived value?
This comment applies in the other `useEffect`s that have `originalNodeObject` as a dependency (in this file and in `ListAllEntryProperties`).